lib/chef/knife/ec2_base.rb

```diff
@@ -207,5 +219,25 @@ def ini_parse(file)
       end
       map
     end
+
+    # All valid platform
+    def valid_platform
+      ["ubuntu", "debian", "centos", "fedora", "rhel", "nginx", "turnkey", "jumpbox", "coreos", "cisco"]
```

Comment:
Do we need to hardcode this list? I can see a number of future issues filed to add to it.